test/mocks/MockBadSubscribers.sol

```diff
@@ -56,3 +56,45 @@ contract MockReturnDataSubscriber is ISubscriber {
         memPtr = _value;
     }
 }
+
+/// @notice A subscriber contract that returns values from the subscriber entrypoints
+contract MockRevertSubscriber is ISubscriber {
+    PositionManager posm;
+
+    error NotAuthorizedNotifer(address sender);
+
+    error TestRevert(string);
```

Comment:
would be nice to see different error types like string errors / requires and simple errors (parameter-less)